src/Simulations/run.jl

```diff
@@ -94,7 +94,7 @@ function run!(sim; pickup=false)
     start_run = time_ns()
 
     if we_want_to_pickup(pickup)
-        set!(sim, pickup)
+        pickup!(sim, pickup)
```

Comment:
isn't this just changing the name of `set!` to `pickup!`?

Reply:
yes, well for an ocean model in Oceananigans pickup! is just that set!, but in ClimaOcean pickup! does more than that

Reply:
By exposing the pickup! function, we are able to define a OceanSeaIceModel specific pickup!:

```
function pickup!(sim::OSIMSIM, pickup)
    @info "i try to properly pick up"
    set!(sim.model.ocean, pickup)
    return nothing
end
```

Reply:
can you extend `set!` for `OSIMSIM` to achieve the same thing?

Reply:
```julia
function set!(sim::OSIMSIM, pickup)
    @info "i try to properly pick up"
    set!(sim.model.ocean, pickup)
    return nothing
end
```

Reply:
This is a lot less work than modifying `set!` itself, as that would require changing `write_output!` and `checkpointer_address`, among others

Reply:
I know I suggested this strategy 😭 I didn't realize that it was _already_ exposed neatly

Reply:
OK, I manage to make it work without `pickup!`!